src/common/transformations/tests/common_optimizations/shared_ops_optimization.cpp

```diff
@@ -433,3 +433,78 @@ TEST_F(SharedTransformationTestsF, SharedShapeOfTestMixed) {
         model_ref = std::make_shared<Model>(NodeVector{concat}, ParameterVector{input});
     }
 }
+
+namespace {
+template <typename T, typename... NodeArgTypes>
+OutputVector createNodesInMemory(const std::vector<size_t>& node_order_in_memory,
+                                 std::shared_ptr<void>& memory,
+                                 const std::string& node_name_prefix,
+                                 NodeArgTypes&&... node_args) {
+    OutputVector outputs;
+    memory.reset(::malloc(node_order_in_memory.size() * sizeof(T)), ::free);
+    for (size_t i = 0; i < node_order_in_memory.size(); ++i) {
+        T* node_addr = static_cast<T*>(memory.get()) + node_order_in_memory[i];
```

Comment:
are T* and static_cast<T*> necessary here?

Reply:
This is to avoid warning: "pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]"